src/private/lidar_impl.cpp

```diff
@@ -0,0 +1,191 @@
+#include "context_impl.cpp"
+#include <vector>
+#include <libm2k/utils/utils.hpp>
+#include <libm2k/utils/channel.hpp>
+#include <libm2k/utils/devicegeneric.hpp>
+#include <libm2k/utils/devicein.hpp>
+#include <libm2k/lidar.hpp>
+#include <iio.h>
+#include <iostream>
+
+using namespace std;
+using namespace libm2k::contexts;
+using namespace libm2k::utils;
+
+class Lidar::LidarImpl : public ContextImpl {
+public:
+	LidarImpl(std::string uri, iio_context* ctx, std::string name, bool sync) :
+		ContextImpl(uri, ctx, name, sync) {
+
+		pulse_dev = new DeviceGeneric(ctx, "7c700000.axi-pulse-capture");
+		afe_dev	  = new DeviceGeneric(ctx, "ad5627");
+		adc_dev	  = new DeviceIn     (ctx, "axi-ad9094-hpc");
+
+		setAllIIoAttributesToDefaultValues();
+	}
+
+	void initialize() {
+		;
+	}
+    
+	void channelEnableDisable(string channel, bool enable) {
+		if	(channel == "voltage0")
+			adc_dev->enableChannel(0, enable, false);
+		else if (channel == "voltage1")
+			adc_dev->enableChannel(1, enable, false);
+		else if (channel == "voltage2")
+			adc_dev->enableChannel(2, enable, false);
+		else if (channel == "voltage3")
+			adc_dev->enableChannel(3, enable, false);
+		else if (channel == "voltage4")
+			adc_dev->enableChannel(4, enable, false);
+		else
+			return;		    // invalid channel
```

Comment:
should this signal an error?
or through an exception?

Reply:
Nothing happens if you try to enable a channel that does not exist. Looks clean to me.

Reply:
Should we let the user know he didn't do something right? 
My opinion is that we could throw here something like: throw_exception(EXC_INVALID_PARAMETER, "Lidar: Can't find channel" + channel);
Might be useful? 
For example: If the user has a typo in the channel name, he'll think that he enabled one channel, when in fact, nothing happened.

Reply:
Depends on what the general approach is in case the error is on the user side, meaning, it's not using the library as it should be used (as I've happened to do countless times). Specifying a wrong channel, even though the user has access to all the available channels, is a user error, not a libm2k error.

If you insist on implementing with throw, I can do that.

Reply:
> Depends on what the general approach is in case the error is on the user side, meaning, it's not using the library as it should be used [snip]

the question is more like: if someone were to use this library as-is: what would he/she expect to see/get if something doesn't go correctly?
1. no error?
2. a return code?
3. a C++ exception?
4. something else?

my sentiment is that this is an API-style library; and these things should be accounted for;

Reply:
I think in this case we should throw an exception. 
We don't really use return codes right now and we want the user to know there's something wrong with the way he's using this.

Reply:
Fixed by throwing an exception.